src/ShipAICmd.cpp

```diff
@@ -992,9 +1035,47 @@ bool AICmdFlyTo::TimeStepUpdate()
 	FrameId targframeId = m_target ? m_target->GetFrame() : m_targframeId;
 	ParentSafetyAdjust(m_dBody, targframeId, targpos, targvel);
 	vector3d relpos = targpos - m_dBody->GetPosition();
+	double targdist = relpos.Length();
+
+	Body* planetNeer = Frame::GetFrame(m_dBody->GetFrame())->GetBody();
+	double safeAlt = MaxEffectRad(planetNeer, m_prop);
+	vector3d obspos = -m_dBody->GetPosition();
+
+	double targetAlt = targpos.Length();
+
+	if ((m_suicideRecovery = CheckSuicide(m_dBody, obspos, planetNeer->GetMass(), safeAlt, targetAlt, m_suicideRecovery))) {
```

Comment:
`planetNeer->GetMass()`

This causes an assertion failure due to the Body for a spaceport's frame being the spaceport itself and not the planet. You will need to check `planetNeer->IsType(ObjectType::TERRAINBODY)` before calling GetMass().

(Also, `planetNeer` should probably be `planetNear`.)

Reply:
Corrected